src/libs/ReportActionsUtils.ts

```diff
@@ -635,6 +665,147 @@ function isNotifiableReportAction(reportAction: OnyxEntry<ReportAction>): boolea
+            return {content: `@${displayNameOrLogin}`, accountID} as MemberChangeMessageUserMentionElement;
+        }
+        return {content: `@${Localize.translateLocal('common.hidden')}`, accountID} as MemberChangeMessageUserMentionElement;
+    });
+
+    const lastMention = mentions.pop();
+    if (!lastMention) {
+        return [];
+    }
+
+    if (mentions.length === 0) {
+        messageItems.push({
+            kind: 'userMention',
+            content: lastMention.content,
+            accountID: lastMention.accountID,
+        });
+    } else if (mentions.length === 1) {
+        messageItems.push(
+            {
+                kind: 'userMention',
+                content: mentions[0].content,
+                accountID: mentions[0].accountID,
+            },
+            {
+                kind: 'text',
+                content: ` ${Localize.translateLocal('common.and')} `,
+            },
+            {
+                kind: 'userMention',
+                content: lastMention.content,
+                accountID: lastMention.accountID,
+            },
+        );
+    } else {
+        mentions.forEach((mention) => {
+            messageItems.push(
+                {
+                    kind: 'userMention',
+                    accountID: mention.accountID,
+                    content: `${mention.content}`,
+                },
+                {
+                    kind: 'text',
+                    content: `, `,
+                },
+            );
+        });
+
+        messageItems.push(
+            {
+                kind: 'text',
+                content: `${Localize.translateLocal('common.and')} `,
+            },
+            {
+                kind: 'userMention',
+                content: lastMention.content,
+                accountID: lastMention.accountID,
+            },
+        );
+    }
+
+    const roomName = originalMessage?.roomName;
+    if (roomName) {
+        const preposition = isInviteAction ? ` ${Localize.translateLocal('workspace.invite.to')} ` : ` ${Localize.translateLocal('workspace.invite.from')} `;
+
+        if (originalMessage.reportID) {
+            messageItems.push(
+                {
+                    kind: 'text',
+                    content: preposition,
+                },
+                {
+                    kind: 'roomReference',
+                    roomName,
+                    roomID: originalMessage.reportID,
+                    content: roomName,
+                },
+            );
+        }
+    }
+
+    return messageItems;
+}
+
+function getMemberChangeMessageFragment(reportAction: OnyxEntry<ReportAction>): Message {
+    const messageItems: MemberChangeMessageElement[] = getMemberChangeMessageElements(reportAction);
+    let html = '';
+
+    messageItems.forEach((messageItem) => {
```

Comment:
Isn't this `forEach` logic directly equivalent to a pure expression using the `join` operator?

Reply:
`map` +`join`?